### ISSUE TITLE
A11y keyboard focus control for Sidenav Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3603]
#### Bug, feature request, or proposal:
Bug:
The "sidenav" component lacks page structure for content context relationships.

#### What is the expected behavior?
The screen reader focus should always land at the top of the Sidenav content, and should be able to distinguish Sidenav content from other page content.

#### What is the current behavior?
Currently pressing Spacebar on the "add" button appears to do nothing, as there is no feedback, and the text is not identified as part of the sidenav.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Creating a good keyboard user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer, and Google Chrome.

#### Is there anything else we should know?
The screen reader user experience follows:

1. Press the Tab key to move focus to the example sidenav "add button".
When using Firefox JAWS says "add button" followed by the latin text, and the braille display shows "add btn" (Note, only the element with focus should be spoken).
When using IE11 JAWS says "add image", and displays "add img" (Note, the button and text are in the focus order twice).
The sidenav content is not enclosed in any identifiable Landmark region. This makes it difficult to understand content relationships with other page content.

2. Activate the Add button.
Pressing Spacebar or Enter key on the "add button" simply duplicates the sidenav (now two copies of the "add button" and the latin text). However, there is no focus change or screen reader feedback to let the user know some action has been taken.
Developers should be encouraged to structure the page content and widgets with HTML structure, and where necessary ARIA markup.
First, enclose the sidenav in an ARIA region;
role=region aria-label="sidenav"  (where "sidenav" is brief text to describe the region, such as "Advertisement", "Social Media", "Special Offers", "Demonstrations", ...).
If the sidenav is a navigation area;
role=navigation aria-label="Site"  (for site navigation).
Secondly, the sidenav should have a Heading at the top of the content that follows the hierarchy structure. If a Heading is used then the Heading text can be used to identify the region;
role=region aria-labelledby=hdrtext
<h2 id=hdrtext>Side Nav Heading</h2>
Note, there are many ARIA "role" types for different page sections, but they should always be used with some text description.
